### PR TITLE
Add Algolia for searching users in NavBar

### PR DESCRIPTION
**Pull request by dmitrymarchuk**
## Aglolia search

This PR adds [Algolia](https://www.algolia.com) search to the users search in `NavBar`. 
Algolia is a hosted search engine. First, all the documents of the given firestore collection must be indexed, and then requests to the Algolia servers can be performed.
Algolia Firestore extension adds a cloud function `ext-firestore-algolia-search-executeIndexOperation` which automatically updates search index after every doc change in `users` collection. 

## Indexation
Initial indexation or (re-indexation after configuration changes) is performed with this bash command:
```bash
LOCATION=us-central1 \
PROJECT_ID=co-reality-staging \
ALGOLIA_APP_ID= \
ALGOLIA_API_KEY= \
ALGOLIA_INDEX_NAME=algolia-users \
COLLECTION_PATH=users \
FIELDS=partyName,pictureUrl,anonMode \
GOOGLE_APPLICATION_CREDENTIALS= \
npx firestore-algolia-search
```

`PROJECT_ID` - firebase project ID where extension is added.
`ALGOLIA_APP_ID`, `ALGOLIA_API_KEY` - can be found in Algolia's [dashboard](https://www.algolia.com/api-keys).
`ALGOLIA_INDEX_NAME` - a name of an index created in Algolia's [dashboard](https://www.algolia.com/explorer).
`COLLECTION_PATH` and `FIELDS` - what should be indexed from a given firestore collection
`GOOGLE_APPLICATION_CREDENTIALS` - a private key for [Google Application Default Credentials](https://firebase.google.com/docs/admin/setup#initialize-sdk)

## Configuration

Developers must add a `REACT_APP_ALGOLIA_API_SEARCH_KEY` to the `.env.local`

## PR changes overview

- Creates Algolia `Context` and adds `useAlgoliaSearch` hook for performing search requests.
- Removes `worldUsers` usage from `NavBar`
- Updates `UserAvatar` types to only use a subset of `User` fields.

### CODE REVIEW

File: src/hooks/algolia/context.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  const client = algoliasearch(appId, key);
+  const indices = Object.assign(
+    {},
+    ...algoliaSearchIndexes.map((indexName: AlgoliaSearchIndex) => {
+      const index = client.initIndex(indexName);
+      return {
+        [indexName]: index,
+      };
+    })
+  );
```
Comment (mike-lvov) [inline src/hooks/algolia/context.tsx:30]: There is a lodash method that turns arrays into objects. Please, use that

File: src/hooks/algolia/useAlgoliaSearch.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            "enteredVenueIds"
+          )}: ${venueId}`,
+        },
+      }))
+    );
+
+    return Object.assign(
+      {},
+      ...indexes.map((indexName, i) => ({ [indexName]: results[i] }))
+    );
```
Comment (mike-lvov) [inline src/hooks/algolia/useAlgoliaSearch.ts:36]: Same here as above ^

**Review by mike-lvov — APPROVED** [review]
Looks good with a few updates

File: src/hooks/algolia/context.tsx (hunk 1 above)
Comment (dmitrymarchuk) [reply src/hooks/algolia/context.tsx]: fixed in https://github.com/sparkletown/sparkle/pull/2285/commits/591237fbb78709a84f54371437645023748f2b54

File: src/hooks/algolia/useAlgoliaSearch.ts (hunk 2 above)
Comment (dmitrymarchuk) [reply src/hooks/algolia/useAlgoliaSearch.ts]: fixed in https://github.com/sparkletown/sparkle/pull/2285/commits/591237fbb78709a84f54371437645023748f2b54

**Pull request merged**